intern/cycles/kernel/kernel_emission.h

```diff
@@ -126,7 +125,7 @@ ccl_device_noinline_cpu bool direct_emission(KernelGlobals *kg,
 
 #ifdef __VOLUME__
   if (sd->prim != PRIM_NONE)
-    shader_bsdf_eval(kg, sd, ls->D, eval, ls->pdf, ls->shader & SHADER_USE_MIS);
+    shader_bsdf_eval(kg, sd, ls->D, eval, ls->pdf, ls->shader & SHADER_USE_MIS, state->wavelengths);
```

Comment:
Anywhere you see `state->wavelengths` being passed, it's just because that function or something inside of it needs access to the wavelengths (most likely for converting RGB into a spectrum in the BSDF evaluation). I wish there was a nicer way to do this, and I think there will be after the `next big hurdle`.

Reply:
The next big hurdle is to move all the spectrum conversions out of each BSDF and instead do it just as RGB data gets into the system, that way it won't be the responsibility of each BSDF to do it (which has been pretty tedious) and instead will just happen once in an RGB node, image node, or something of that type. I'm still not entirely clear on how I'm going to tackle this, but it needs to happen in order to be able to have nodes in the material system 'communicate' with spectra.